lnrpc/peersrpc/peers_server.go

```diff
@@ -292,7 +281,13 @@ func (s *Server) updateFeatures(currentfeatures *lnwire.RawFeatureVector,
 		}
 	}
 
-	// Validate our new SetNodeAnn.
+	raw, err := lnwire.MergeFeatureVector(
+		currentfeatures, add, remove, s.cfg.ConfigFeatures,
+	)
```

Comment:
I think you could call this a more 'external' approach to managing the feature bits. Would this have been possible as an extension of `feature.Manager` too? Perhaps that would keep the logic a bit more in one place.

Reply:
Going to take a look at moving this into the feature manager. The existing API only advertises custom features in gossip (not peer init), but I think that it makes sense to update it to do both. Seems like it'll simplify things a bit as well 🤞